### PR TITLE
Magento 2.4.1  and 2.3.6 Pre-release Release Notes

### PR DESCRIPTION
**Pull request by jfrontain**
## Purpose of this pull request

This pull request (PR) adds  drafts of the 2.3.6, 2.4.1, and B2B release notes. These drafts are incomplete and subject to change and additions.

### CODE REVIEW

File: src/guides/v2.4/release-notes/commerce-2-4-1.md [hunk 1]
```diff
@@ -208,48 +208,74 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
 <!--- ENGCOM-7459-->
 
-<!--- ENGCOM-7265-->
+*  Storefront performance has improved by eliminating the unnecessary loading  of the `Datepicker` component. _Fix submitted by Mateusz Krzeszowiak in pull request [27860](https://github.com/magento/magento2/pull/27860)_. [GitHub-28823](https://github.com/magento/magento2/issues/28823)
 
-<!--- ENGCOM-7508-->
+<!--- ENGCOM-7265-->
 
-<!--- ENGCOM-7800-->
+*  Executing `php bin/magento s:up` now completes as expected. Previously, Magento displayed printed array content for caches. _Fix submitted by Sathish Subramanian in pull request [27567](https://github.com/magento/magento2/pull/27567)_. [GitHub-27091](https://github.com/magento/magento2/issues/27091)
```
Comment (hostep) [inline src/guides/v2.4/release-notes/commerce-2-4-1.md:227]: We should change `s:up` to `setup:upgrade`, that's the more official command :)

File: src/guides/v2.4/release-notes/commerce-2-4-1.md [hunk 2]
```diff
@@ -208,48 +208,74 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
 
 <!--- ENGCOM-7629-->
 
-<!--- ENGCOM-7776-->
+*  Running `./bin/magento config:show vendor_module/general/value` now returns `0` or '' (an empty string) as expected. Previously, this command returned `Configuration for path: "vendor_module/general/value" doesn't exist`. _Fix submitted by Vadim Malesh in pull request [28549](https://github.com/magento/magento2/pull/28549)_. [GitHub-23290](https://github.com/magento/magento2/issues/23290)
```
Comment (hostep) [inline src/guides/v2.4/release-notes/commerce-2-4-1.md:231]: Is already mentioned a couple of paragraphs above, best to merge it.

File: src/guides/v2.4/release-notes/commerce-2-4-1.md [hunk 3]
```diff
@@ -683,27 +858,123 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
 
 <!--- ENGCOM-7073-->
 
-<!--- MC-39468-->
+*  `Magento_CacheInvalidate` now handles large tag patterns correctly when doing a `PURGE`. `sendPurgeRequest` has been refactored to handle an array of tags instead of requiring the caller to use `implode()`.
 
 <!--- MC-30568-->
 
-*  Shared indexers now show a status of **valid** after you run `bin/magento indexer:status` after re-indexing. Previously, shared indexers had an **invalid** status after a full re-index.
+*  Shared indexers now show a status of **valid** when you run `bin/magento indexer:status` after re-indexing. Previously, shared indexers had an **invalid** status after a full re-index.
```
Comment (hostep) [inline src/guides/v2.4/release-notes/commerce-2-4-1.md:883]: This fix was already included in 2.4.0, so this should get moved to the 2.4.0 release notes where it seems to be missing.

File: src/guides/v2.4/release-notes/commerce-2-4-1.md [hunk 4]
```diff
@@ -683,27 +858,123 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
+
+*  `ResourceConnection.php` has been refactored to improve class readability. _Fix submitted by Lukasz Bajsarowicz in pull request [29341](https://github.com/magento/magento2/pull/29341)_. [GitHub-29389](https://github.com/magento/magento2/issues/29389)
+
+<!--- ENGCOM-7910-->
+
+*  The README file for the build-in web server has been updated to include all Elasticsearch parameters. _Fix submitted by Yevhenii Dumskyi in pull request [29300](https://github.com/magento/magento2/pull/29300)_. [GitHub-29299](https://github.com/magento/magento2/issues/29299)
+
+<!--- ENGCOM-7814-->
+
+*  All after plugins now return a value as expected. Previously, `Magento\CmsUrlRewrite\Plugin\Cms\Model\Store\View::aftersSave` did not return a value, and as a result, saving a store view resulted in an error. _Fix submitted by Pieter Hoste in pull request [29035](https://github.com/magento/magento2/pull/29035)_. [GitHub-29034](https://github.com/magento/magento2/issues/29034)
```
Comment (hostep) [inline src/guides/v2.4/release-notes/commerce-2-4-1.md:953]: I didn't check *all* after plugins, so this statement could be incorrect. I only fixed this one after plugin which gave us errors in one particular shop 🙂

**Review by hostep — COMMENTED** [review]
Only very quickly went through it and found some issues here and there.

File: src/guides/v2.4/release-notes/commerce-2-4-1.md (hunk 3 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/commerce-2-4-1.md:883]: I've removed it from these notes and will add it to the 2.4.0 notes in a separate PR. I'm glad you caught this -- the fix version was incorrect in JIRA, which is why my query picked it up.

File: src/guides/v2.4/release-notes/commerce-2-4-1.md (hunk 2 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/commerce-2-4-1.md:231]: merged. thanks. I need o watch out for fixes that tickets in both the ENGCOM and MC projects.

File: src/guides/v2.4/release-notes/commerce-2-4-1.md (hunk 1 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/commerce-2-4-1.md:227]: done

**Review by keharper — CHANGES_REQUESTED** [review]
I only reviewed the 2.4.1 Open Source release notes and EE-specific sections in the 2.4.1 Commerce release notes.

File: src/guides/v2.4/release-notes/commerce-2-4-1.md [hunk 5]
```diff
@@ -713,18 +976,58 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
 
 <!--- ENGCOM-7493-->
 
+*  The swatch layered navigation filter is now consistent with standard filters used throughout Magento. _Fix submitted by Bartłomiej Szubert in pull request [28015](https://github.com/magento/magento2/pull/28015)_. [GitHub-28011](https://github.com/magento/magento2/issues/28011)
+
+### Logging
+
+<!--- ENGCOM-7692-->
+
+*  All broken reference errors are now logged when deployments are in developer mode only. Previously, one error was logged for deployments in production mode, too, which bloated error logs. _Fix submitted by Bart��omiej Szubert in pull request [28735](https://github.com/magento/magento2/pull/28735)_. [GitHub-26504](https://github.com/magento/magento2/issues/26504)
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/commerce-2-4-1.md:993]: searched and replaced. thanks.

File: src/guides/v2.4/release-notes/open-source-2-4-1.md [hunk 6]
```diff
@@ -591,9 +1206,37 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
+
+*  Adds a test for deleting items from a customer’s wish list as an Admin user. _Fix submitted by Dmitry Tsymbal in pull request [28632](https://github.com/magento/magento2/pull/28632)_. [GitHub-28721](https://github.com/magento/magento2/issues/28721)
+
+<!--- ENGCOM-7675-->
+
+*  Adds a test for disabling wish list functionality. _Fix submitted by Dmitry Tsymbal in pull request [28635](https://github.com/magento/magento2/pull/28635)_. [GitHub-28744](https://github.com/magento/magento2/issues/28744)
+
+<!--- ENGCOM-7564-->
+
+*  Magento now correctly validates the allowed maximum of wish lists that a shopper can create. _Fix submitted by Eduard Chitoraga in pull request [247](https://github.com/magento/partners-magento2ee/pull/247)_. [GitHub-261](https://github.com/magento/partners-magento2ee/issues/261)
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-1.md:1243]: done

File: src/guides/v2.4/release-notes/open-source-2-4-1.md [hunk 7]
```diff
@@ -140,45 +142,132 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
 
 <!--- MC-35001-->
 
 *  The **Use default** checkbox for Klarna payments (**Stores** > **Configuration** > **Sales** > **Payment methods** > **Klarna**) now remain checked as expected when website scope changes.
 
-<!--- MC-17959-->
+<!--- MC-33788-->
 
-*  Running `/bin/magento config:show vendor_module/general/value` now returns `0` or an empty string as expected. Previously, it returned `Configuration for path: "vendor_module/general/value" doesn't exist`. [GitHub-23290](https://github.com/magento/magento2/issues/23290)
+*  Upgrade no longer results in the sudden failure of the Galera cluster. Previously, the Galera cluster exited abruptly after re-indexing immediately after upgrade. During Magento upgrade, index tables are altered, and the engine is changed from `MEMORY` to `InnoDB`. At this point, the content of these tables becomes out-of-sync between the nodes of the Galera cluster. [GitHub-25334](https://github.com/magento/magento2/issues/25334)
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-1.md:153]: done. might edit further.

File: src/guides/v2.4/release-notes/open-source-2-4-1.md [hunk 8]
```diff
@@ -328,7 +609,63 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
+
+*  Saving an attribute with `backend_type = static` no longer removes the content of the `frontend_class` field. _Fix submitted by jiten-patel in pull request [27369](https://github.com/magento/magento2/pull/27369)_. [GitHub-27051](https://github.com/magento/magento2/issues/27051)
+
+<!--- ENGCOM-7462-->
+
+*  Unnecessary code and `responsive.js` have been removed from files that are loaded by themes. _Fix submitted by Mateusz Krzeszowiak in pull request [27617](https://github.com/magento/magento2/pull/27617)_. [GitHub-28811](https://github.com/magento/magento2/issues/28811)
+
+<!--- ENGCOM-7729-->
+
+*  Category queries now return correct response when using inline fragments. _Fix submitted by Ulzii in pull request [28710](https://github.com/magento/magento2/pull/28710)_. [GitHub-28584](https://github.com/magento/magento2/issues/28584)
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-1.md]: have removed and will include in GraphQL section next draft

File: src/guides/v2.4/release-notes/open-source-2-4-1.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-1.md:1243]: deleted in OS

File: src/guides/v2.4/release-notes/commerce-2-4-1.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/commerce-2-4-1.md:993]: replaced as needed

File: src/guides/v2.4/release-notes/open-source-2-4-1.md [hunk 9]
```diff
@@ -354,27 +709,161 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
+
+*  Array creation is now consistent throughout the class (`app/code/Magento/Sales/Model/Order/Pdf/Items/Invoice/DefaultInvoice.php`). _Fix submitted by Nathan de Graaf in pull request [28515](https://github.com/magento/magento2/pull/28515)_. [GitHub-28795](https://github.com/magento/magento2/issues/28795)
+
+<!--- ENGCOM-7484-->
+
+*  Plugins have been migrated out of the Magento Framework to follow the Magento best practice of prohibiting plugins in the Framework namespace. _Fix submitted by Lukasz Bajsarowicz in pull request [27965](https://github.com/magento/magento2/pull/27965)_. [GitHub-27962](https://github.com/magento/magento2/issues/27962)
+
+<!--- ENGCOM-7817-->
+
+*  GraphQL product search now considers Category Permissions configuration. _Fix submitted by Cristian Partica in pull request [5](https://github.com/magento/partners-magento2-infrastructure/pull/5)_.
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-1.md:768]: removed all references to private repos from these notes, including the include files. Removed issue description from OS notes.

File: src/guides/v2.4/release-notes/open-source-2-4-1.md [hunk 10]
```diff
@@ -447,17 +958,25 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
 <!--- MC-33952-->
 
-*  Elasticsearch results now include the correct values for each storeview’s attribute options. If a Dropdown or Multiple Select attribute has a different option value in the non-default store view than in the default storeview, Elasticsearch now indexes that value or returns the product with that value in the results. Previously, Elasticsearch  did not index that value or return the product with that value in the results.
+*  Elasticsearch results now include the correct values for each store view’s attribute options. If a Dropdown or Multiple Select attribute has a different option value in the non-default store view than in the default store view, Elasticsearch now indexes that value or returns the product with that value in the results. Previously, Elasticsearch did not index that value or return the product with that value in the results.
 
 <!--- MC-35013-->
 
 *  Searching by SKU now works as expected in advanced search with Elasticsearch 6. Previously, when you tried to search by SKU, Magento displayed this error message: `We can't find any items matching these search criteria. Modify your search`.
 
+<!--- ENGCOM-7222-->
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-1.md:975]: right. corrected.

File: src/guides/v2.4/release-notes/open-source-2-4-1.md [hunk 11]
```diff
@@ -409,6 +898,24 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
    *  Network cache size has decreased by 10 - 15%
    *  Execution time for many scenarios has been improved by 3%.
 
+<!--- ENGCOM-7290-->
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-1.md:909]: added to both files

File: src/guides/v2.4/release-notes/open-source-2-4-1.md [hunk 12]
```diff
@@ -220,24 +363,118 @@ We have fixed hundreds of issues in the Magento 2.4.1 core code.
...
+<!--- ENGCOM-7156-->
+
+*  Magento now creates a maximum of one `mage-translation-file-version` and `mage-translation-storage` cookie per session. _Fix submitted by Ihor Sviziev in pull request [27364](https://github.com/magento/magento2/pull/27364)_. [GitHub-27355](https://github.com/magento/magento2/issues/27355)
+
+### cron
+
+<!--- MC-35884-->
+
+*  Message queue consumer configuration has been extended with new parameters that help control consumers and save server resources and that potentially decrease consumer queue CPU consumption by 20%:
+
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-1.md:463]: edited

**Review by chernenm — APPROVED** [review]

**Review by keharper — APPROVED** [review]

**Comment by jfrontain** [comment]
running tests

**Pull request merged**